proxy.go

```diff
@@ -51,9 +51,12 @@ func (ph ProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func (ph ProxyHandler) handleConnect(w http.ResponseWriter, req *http.Request) {
+	// Establish a connection to the server, or an upstream proxy.
 	u, err := ph.transport.Proxy(req)
+	id := req.Context().Value("id")
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("[%d] Can't find proxy for %v: %v", id, req.Host, err)
```

Comment:
Most of the other errors messages start with "Error". Should you standardise on that for errors being logged? If so this could be "Error finding proxy for..."

Reply:
Done